internal/engine/compiler/compiler_test.go

```diff
@@ -242,6 +244,22 @@ func (j *compilerEnv) exec(codeSegment []byte) {
 	)
 }
 
+func (j *compilerEnv) execBench(codeSegment []byte, b *testing.B) {
+	f := j.newFunctionFrame(codeSegment)
+
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		j.ce.callFrameStack[j.ce.globalContext.callFrameStackPointer] = callFrame{function: f}
+		j.ce.globalContext.callFrameStackPointer++
+		nativecall(
+			uintptr(unsafe.Pointer(&codeSegment[0])),
+			uintptr(unsafe.Pointer(j.ce)),
+			uintptr(unsafe.Pointer(j.moduleInstance)),
+		)
```

Comment:
we should check the result in each loop, rather than after the timer.

Reply:
What do we benchmark then 🤔 ? We'll have a lot of interference from go functions (copy, equal)

Reply:
no these are not interference at at all, but it depends how expensive they are. What we should really care is the case like this: the logic only peformes correctly at 1000th iteration, and the other cases, it just does the NOP in the loop. That would be wrong benchmarks result and that's the _really_ common mistake in benchmark.

Reply:
the other way you can do is to write unit tests on the logic performed in the each loop iteration -- you can extract it as the separate function, and then write unit test for it.